cMake/FreeCAD_Helpers/SetGlobalCompilerAndLinkerSettings.cmake

```
@@ -19,6 +19,14 @@ macro(SetGlobalCompilerAndLinkerSettings)
         set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mxgot")
     endif()
 
+    if (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+        set (CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -Og")
+        set (CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -Og")
+    elseif (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+        set (CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -Og")
+        set (CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -Og")
+    endif()
+
```

If a user specifies an optimization level manually, like bof mentioned in the other PR, wouldn't we want to keep that setting instead of appending this? (`... -O0 -Og` would override it).

Maybe something like this would solve that problem:
 ```suggestion
if (CMAKE_CXX_COMPILER_ID MATCHES "Clang|GNU")
   # Only add -Og if no -O* optimization flag exists
   if (NOT "${CMAKE_C_FLAGS_DEBUG}" MATCHES "-O[a-z0-9]+")
       set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -Og")
   endif()
   if (NOT "${CMAKE_CXX_FLAGS_DEBUG}" MATCHES "-O[a-z0-9]+")
       set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -Og")
   endif()
endif()

```